django/core/signing.py

```diff
@@ -135,18 +144,19 @@ def dumps(obj, key=None, salt='django.core.signing', serializer=JSONSerializer,
     return TimestampSigner(key, salt=salt).sign_object(obj, serializer=serializer, compress=compress)
 
 
-def loads(s, key=None, salt='django.core.signing', serializer=JSONSerializer, max_age=None):
+def loads(s, key=None, salt='django.core.signing', serializer=JSONSerializer, max_age=None, read_keys=None):
     """
     Reverse of dumps(), raise BadSignature if signature fails.
 
     The serializer is expected to accept a bytestring.
     """
-    return TimestampSigner(key, salt=salt).unsign_object(s, serializer=serializer, max_age=max_age)
+    return TimestampSigner(key, salt=salt, read_keys=read_keys).unsign_object(s, serializer=serializer, max_age=max_age)
 
 
 class Signer:
-    def __init__(self, key=None, sep=':', salt=None, algorithm=None):
+    def __init__(self, key=None, sep=':', salt=None, algorithm=None, read_keys=None):
         self.key = key or settings.SECRET_KEY
+        self.read_keys = read_keys or list(settings.SECRET_KEY_FALLBACKS)
```

Comment:
Why the `list()` here? (Is `settings.SECRET_KEY_FALLBACKS` enforced as a sequence by the system check?)

Reply:
I think I used list here due to my inexperience with the Django codebase. I'll get that removed.

Reply:
From the rest of the changes, I think you've coerced it intentionally so you can do `[self.key] + self.read_keys` (although at a glance there's also `[self.secret] + self.secret_fallbacks` which isn't enforced, at a glance?)

Reply:
Yes, I'm wrong in at least one spot.